aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -211,6 +227,40 @@ namespace {
     }
     return descriptors;
   }
+#else
+  auto rnn_descriptor_sequence(const Tensor& tensor, const int batch_size, IntArrayRef batch_sizes, const int seq_len, const int vector_size) { // packed case
+    RNNDataDescriptor r;
+    std::vector<int> seqLengthArray(batch_size, 1);
+    // cuDNN wants the sequence lenghts for a packed batch as if they
+    // were unpacked, e.g., for the
+    // Sequence 1: ABCD
+    // Sequence 2: EF
+    // Sequence 3: G
+    // case below, this would be [4, 2, 1] (has length == mini_batch)
+    // TODO(eqy): There's probably a smarter way to do this than O(SN)
+    for (auto it = batch_sizes.begin(); it != batch_sizes.end(); it++) {
+      // everyone starts at sequence length 1 so we skip an iteration
+      if (it == batch_sizes.begin()) {
+        continue;
+      }
+      for (int idx = 0; idx < *it; idx++) {
+        seqLengthArray[idx]++;
+      }
+    }
+    r.set(tensor, CUDNN_RNN_DATA_LAYOUT_SEQ_MAJOR_PACKED, seq_len, batch_size, vector_size, seqLengthArray.data());
+    return r;
+  }
+
+  auto rnn_descriptor(const Tensor& tensor, const int batch_size, const int seq_len, const int vector_size) {
```

Comment:
Two nits: can those be unsigned values (i.e. vector size can't really be negative, and same I assume is batch size and everything else. Also, why do they need to be const?
```suggestion
  auto rnn_descriptor(const Tensor& tensor, uint32_t batch_size, uint32_t seq_len, uint32_t vector_size) {
```

Reply:
I was taking the approach of making things `const` unless proven otherwise---will take another pass making this more consistent.